### PR TITLE
add program-date-time tag info to parsed segments

### PR DESCRIPTION
**Pull request by mjneil**
See #26 

This adds program date time info onto the parsed segment objects as well as the manifest since it is meant to be a segment tag

### CODE REVIEW

File: src/parser.js [hunk 1]
```diff
@@ -267,8 +267,16 @@ export default class Parser extends Stream {
...
             'program-date-time'() {
-              this.manifest.dateTimeString = entry.dateTimeString;
-              this.manifest.dateTimeObject = entry.dateTimeObject;
+              if (typeof this.manifest.dateTimeString === 'undefined') {
+                // PROGRAM-DATE-TIME is a media-segment tag, but for backwards
+                // compatibility, we add the first occurance of the PROGRAM-DATE-TIME tag
+                // to the manifest object
+                this.manifest.dateTimeString = entry.dateTimeString;
+                this.manifest.dateTimeObject = entry.dateTimeObject;
+              }
```
Comment (mister-ben) [inline src/parser.js]: This is needed for calculating stream time from currentTime, so is more important than "for backwards compatibility" implies.

Comment (mjneil) [inline src/parser.js]: You could still calculate stream time from currentTime by looking at the `dateTimeString` attached to the individual segment objects, the backwards compatibility is so updating to a version including this change wont break code that is expecting that information to be on the playlist object

File: src/parser.js [hunk 2]
```diff
@@ -267,8 +267,16 @@ export default class Parser extends Stream {
               this.manifest.discontinuityStarts.push(uris.length);
             },
             'program-date-time'() {
-              this.manifest.dateTimeString = entry.dateTimeString;
-              this.manifest.dateTimeObject = entry.dateTimeObject;
+              if (typeof this.manifest.dateTimeString === 'undefined') {
+                // PROGRAM-DATE-TIME is a media-segment tag, but for backwards
+                // compatibility, we add the first occurance of the PROGRAM-DATE-TIME tag
```
Comment (gesinger) [inline src/parser.js]: occurence

File: src/parser.js [hunk 3]
```diff
@@ -267,8 +267,16 @@ export default class Parser extends Stream {
               this.manifest.discontinuityStarts.push(uris.length);
             },
             'program-date-time'() {
-              this.manifest.dateTimeString = entry.dateTimeString;
-              this.manifest.dateTimeObject = entry.dateTimeObject;
+              if (typeof this.manifest.dateTimeString === 'undefined') {
+                // PROGRAM-DATE-TIME is a media-segment tag, but for backwards
```
Comment (gesinger) [inline src/parser.js]: We should add a TODO notice for the next major version to remove this

**Pull request merged**